tests/functional/partial_parsing/test_partial_parsing.py

```diff
@@ -804,3 +805,31 @@ def test_pp_groups(self, project):
         )
         with pytest.raises(ParsingError):
             results = run_dbt(["--partial-parse", "run"])
+
+
+class TestPublicationArtifactAvailable:
```

Comment:
This seems like a lot of things to run in a test for a small functionality, is it possible to write unit for this case instead of functional tests? Totally understand if that's really tricky to do.

Reply:
Yeah, the "test_partial_parsing" file is getting pretty long. Might be a good idea to split it up into smaller pieces. But probably not for this pull request. Maybe we should open another ticket...

Reply:
Unit testing partial parsing in particular may be especially tricky given how dependent it is on reading & writing files.. but I agree that splitting out some of the longer tests into their own pieces would help for debugging / understanding coverage / readability.